src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -922,422 +1427,131 @@ FESurfaceDistanceEvaluator::collectNeighboringPatchElements(int level_number)
-#endif
-                    for (unsigned int i = 0; i < n_nodes; ++i)
-                    {
-                        for (int d = 0; d < NDIM; ++d)
-                        {
-                            X_node[i][d] = (elem->point(i))(d);
-                        }
-                    }
-
-#if !defined(NDEBUG)
+                    bool node_in_patch = true;
+                    for (unsigned int d = 0; d < NDIM; ++d)
                     {
-                        const unsigned int n_qp = qrule_surface->n_points();
-                        TBOX_ASSERT(n_qp == 1);
-                    }
-#endif
-                    unsigned int qp = 0;
-                    for (unsigned int k = 0; k < NDIM - 1; ++k)
-                    {
-                        interpolate(dX_dxi[k], qp, X_node, *dphi_dxi[k]);
-                    }
-                    if (NDIM == 2)
-                    {
-                        dX_dxi[1] = VectorValue<double>(0.0, 0.0, 1.0);
-                    }
-                    N = (dX_dxi[0].cross(dX_dxi[1])).unit();
-
-                    for (int d = 0; d < NDIM; ++d)
-                    {
-                        avg_unit_normal(d) += N(d);
+                        in_patch = node_in_patch && (x_lower[d] <= n(d) && n(d) <= x_upper[d]);
                     }
                 }
 
-                // Take the average normal and normalize it
-                avg_unit_normal /= static_cast<double>(vec_length);
-                avg_unit_normal /= avg_unit_normal.norm();
-
-                // Average the proj point
-                avg_proj /= static_cast<double>(vec_length);
-
-                // Compute the signed distance function.
-                sgn = avg_unit_normal.dot(P - avg_proj) <= 0.0 ? -1.0 : 1.0;
-                (*d_data)(ci) = sgn * min_dist;
+                // Add the element to the patch list if necessary.
+                if (in_patch)
+                {
+                    d_active_neighbor_patch_bdry_elem_map[local_patch_num].push_back(elem);
+                }
             }
         }
     }
-} // computeSignedDistanceSurfaceMesh
+    return;
+} // collectNeighboringPatchElements
+
+void
+FESurfaceDistanceEvaluator::sideElemVolExtractedBdryMesh()
+{
+    // Clear out the data structure.
+    d_side_elem_vol_extracted_bdry_msh.clear();
 
+    // Move the side element unique pointers to the set
+    for (MeshBase::const_element_iterator itr = d_mesh.active_elements_begin(); itr != d_mesh.active_elements_end();
+         ++itr)
+    {
+        Elem* elem = *itr;
+        // Loop over the sides of the element. If it has no
+        // neighbors on a side, then it MUST live on the boundary
+        // of the mesh
+        for (unsigned int s = 0; s < elem->n_sides(); ++s)
+        {
+            if (elem->neighbor_ptr(s) != nullptr) continue;
+            UniquePtr<Elem> side_elem = elem->build_side_ptr(s, /*proxy*/ false);
+            d_side_elem_vol_extracted_bdry_msh.insert(std::move(side_elem));
```

Comment:
Use assertions related to number of nodes and edges here for 2D  and 3D appropriately.